grapher/mapCharts/MapChart.tsx

```diff
@@ -318,10 +322,10 @@ export class MapChart
         this.mapConfig.projection = value
     }
 
-    @computed get formatTooltipValue(): (d: number | string) => string {
+    @computed private get formatTooltipValue(): (d: PrimitiveType) => string {
```

Comment:
What is the switch over to using `PrimitiveType` here for? It also includes `boolean`, which I don't think we will ever show in a tooltip at this point?

Reply:
It's true, `boolean` is not supported in many parts of our code, but I find it tricky to decide what's best. Considering a lot of stuff doesn't handle `boolean`, probably we should remove it as a possible type in our data tables (basically from `PrimitiveType`).

Right now we don't have many (any?) boolean types in production, even though I think they are possible to define in an Explorer. It's probably worthwhile to have eventually, but not without inspecting all code paths that would need to support it.

I'm thinking about (and slowly moving towards) not making any assumptions about what the incoming values are, because we are intending to make Grapher more reusable. 

As for specific context about why `PrimitiveType` is used specifically here – the type comes from the `datum`:

```ts
    @computed private get datum(): OwidVariableRow<PrimitiveType> | undefined {
        return this.mapTable.get(this.mapColumnSlug).owidRows[0]
    }
```

`PrimitiveType` is the most general type, and if we made no assumptions about input, it _can_ technically be a boolean.

It would either be formatted as `"true"` or `"false"`, unless we implement boolean color scales too. We have quite a lot of "yes/no" map charts where the most reasonable value would be a `boolean` if it was supported, here are just a few where it's encoded as a number instead: https://playfair.owid.cloud/admin/test/embeds?tab=map&ids=118,1190,1203,1571,1675,1677,1678,1679,1680,1681,1682,1806,2025,2098,2099,2105,2108,2114

---

Back on topic: the fix that seems reasonable is to completely drop `boolean` from `PrimitiveType` – do you agree?

The alternative is to assume `boolean` can never be passed to a Map chart, which is not true in our current circumstances, we don't enforce that.

Reply:
I vote for removing it from `PrimitiveType` for now.
We may want to add back a `BooleanColumn` at some point in the future, but once we do that we can reconsider and properly check that everything works as expected.

Reply:
Oops, `Boolean` cells are used in `GrapherGrammar` and `ColumnGrammar`, so we can't drop support in `CoreTable`.

This in turn means these boolean values could leak into chart data too, since that can be specified in an Explorer config.

Do you happen to see any easy way out of this? I think making it possible to define booleans in CoreTable but hardcoding `number | string` in the chart leaves a hole for potential type errors that could be caught. 

One option is to apply a filter in `MapChart#transformTable()` to ensure all booleans are dropped, another option is to leave it as it is.

Reply:
It looks like our data_values never used the boolean type though right? I think we'll have to rethink the architecture of data values a little when we switch to a different data model that avoids putting everything into the data_values mysql. As part of this we will probably also get rid of the terrible ambiguity of whether a variable is numeric or string or mixed. When we do that we will know which type everything is and then we can decide if bools are worth the trouble or not (my guess is not). We should then also maybe rethink how grid grammars work, they are a bit odd alltogehter.

Because we haven't used booleans as data values I am inclined to hardcode to number | string. What we could do though is to add a sanity check into transformTable that only runs in dev builds. This way we don't have overhead in prod, we ourselves would notice if we made a mistake in when we think bools may occur and we can clean this up properly once we restructure our data serving pipeline. What do you think?